### PR TITLE
Add support for globally defining nullValueMapMappingStrategy and nullValueIterableMappingStrategy

### LINKED ISSUE #2953: Feature request, additional global defaults
I was wondering if there was a reason that the null value mapping strategies were not added to the Options class in order to override the built-in defaults? If there is no reason preventing this from being done I was hoping it could be added as feature moving forward?

I'm specifically interested in the`nullValueIterableMappingStrategy` and `nullValueMapMappingStrategy` as accepted best practice in this instance is to return empty collections and maps as opposed to null values. Basically I want to set this policy globally instead of via the currently available options.

Cheers, Andy.

### PR DESCRIPTION
**Pull request by filiphr**
Fixes #2953

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/test/nullvaluemapping/NullValueIterableMappingStrategyTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Filip Hrisafov
+ */
+@WithClasses({
+    CarDto.class,
+    Car.class
+})
+public class NullValueIterableMappingStrategyTest {
```
Comment (Zegveld) [inline processor/src/test/java/org/mapstruct/ap/test/nullvaluemapping/NullValueIterableMappingStrategyTest.java:23]: missing the issuekey annotation here I believe.

File: processor/src/test/java/org/mapstruct/ap/test/nullvaluemapping/NullValueMapMappingStrategyTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Filip Hrisafov
+ */
+@WithClasses({
+    CarDto.class,
+    Car.class
+})
+public class NullValueMapMappingStrategyTest {
```
Comment (Zegveld) [inline processor/src/test/java/org/mapstruct/ap/test/nullvaluemapping/NullValueMapMappingStrategyTest.java:23]: missing the issuekey annotation here I believe.

**Review by Zegveld — CHANGES_REQUESTED** [review]
Looks good to me, just a minor thing about missing `IssueKey` annotations at the tests.

**Review by Zegveld — APPROVED** [review]

**Comment by mikybars** [comment]
Somebody could explain why this feature is not present in the latest release?

**Comment by filiphr** [comment]
The feature is not present in 1.5 because the fix version of #2953 is 1.6

**Pull request merged**